src/System.Private.ServiceModel/tools/CertificateCleanup/Program.cs

```diff
@@ -0,0 +1,108 @@
+            RemoveCertificatesFromStore(StoreName.My, StoreLocation.LocalMachine, clean);
+
+            // We have to remove from the LocalMachine store first, since in Windows, we can't remove 
+            // root certificates from the CurrentUser store.
+            RemoveCertificatesFromStore(StoreName.Root, StoreLocation.LocalMachine, clean);
+            RemoveCertificatesFromStore(StoreName.Root, StoreLocation.CurrentUser, clean);
+
+            Console.WriteLine();
+        }
+
+        private static void RemoveCertificatesFromStore(StoreName storeName, StoreLocation storeLocation, bool clean)
+        {
+            Console.WriteLine("  Checking StoreName '{0}', StoreLocation '{1}'", storeName, storeLocation);
+
+            using (X509Store store = new X509Store(storeName, storeLocation))
+            {
+                try
+                {
+                    store.Open(OpenFlags.ReadWrite | OpenFlags.IncludeArchived);
+                }
+                catch (CryptographicException ex)
+                {
+                    StringBuilder exceptionString = new StringBuilder();
+                    exceptionString.AppendFormat("    Error opening StoreName: '{0}' certificate store from StoreLocation '{1}' in ReadWrite mode ", storeName, storeLocation);
+                    if (storeLocation == StoreLocation.LocalMachine)
+                    {
+                        exceptionString.AppendFormat("{0}    In Linux, this is an expected exception as opening a certificate store ReadWrite to LocalMachine isn't yet supported", Environment.NewLine);
+                        exceptionString.AppendFormat("{0}    In Windows, This is usually due to permissions issues if writing to the LocalMachine location", Environment.NewLine);
+                    }
+                    exceptionString.AppendFormat("{0}    Try running the tool with elevated or superuser permissions.", Environment.NewLine);
+
+                    if (s_debug)
+                    {
+                        exceptionString.AppendFormat("{0}    Exception was: {1}", Environment.NewLine, ex);
+                    }
+                    else
+                    {
+                        exceptionString.AppendFormat("{0}    Exception was: {1}", Environment.NewLine, ex.Message);
+                        exceptionString.AppendFormat("{0}    (Stack trace hidden. Run with --debug to see full traces)", Environment.NewLine);
+                    }
+
+                    Console.WriteLine(exceptionString.ToString());
+                }
+
+                foreach (var cert in store.Certificates.Find(X509FindType.FindByIssuerName, CertificateIssuer, false))
+                {
+                    Console.Write("    {0}. Subject: '{1}'", cert.Thumbprint, cert.SubjectName.Name);
+
+                    if (clean)
+                    {
+                        store.Remove(cert);
+                        Console.Write(" ... removed");
```

Comment:
According to MSDN, yes - if *certificate* is null or if there aren't the required permissions. Permissions should be taken care of above where we open the store read/write. 

But per your comment above, I think we should try/catch here too in case so we do the best we can upon cleanup